packages/superset-ui-chart-controls/src/sections/forecastInterval.tsx

```diff
@@ -0,0 +1,123 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { legacyValidateInteger, legacyValidateNumber, t } from '@superset-ui/core';
+import { ControlPanelSectionConfig } from '../types';
+
+export const forecastEnabled = false;
+export const forecastInterval = 0.8;
+export const forecastPeriods = 10;
+export const forecastSeasonalityDaily = null;
+export const forecastSeasonalityWeekly = null;
+export const forecastSeasonalityYearly = null;
+
+export const forecastIntervalControls: ControlPanelSectionConfig = {
+  label: t('Predictive Analytics'),
+  expanded: false,
+  controlSetRows: [
+    [
+      {
+        name: 'forecastEnabled',
+        config: {
+          type: 'CheckboxControl',
+          label: t('Enable forecast'),
+          renderTrigger: false,
+          default: forecastEnabled,
+          description: t('Enable forecasting'),
+        },
+      },
+    ],
+    [
+      {
+        name: 'forecastPeriods',
+        config: {
+          type: 'TextControl',
+          label: t('Forecast periods'),
+          validators: [legacyValidateInteger],
```

Comment:
Why do we use `legacyValidateInteger` instead of `validateInteger`?

Reply:
It is just a code move, no logical changes have been made. I can do refactor later.

Reply:
👍

Reply:
That's a good question. I remember someone long ago said to just juse legacy for now, but I can't remember why. Maybe now is a good time to migrate over and fix whatever potential bugs there are in the new one.

Reply:
Now is not a great moment to introduce risk... I'd rather add that in a follow-up.